code/game/objects/items/melee/baton.dm

```diff
@@ -60,12 +60,19 @@
 
 /obj/item/melee/baton/Initialize(mapload)
 	. = ..()
+	//So we can deactivate batons when the security level changes
+	RegisterSignal(SSsecurity_level, COMSIG_SECURITY_LEVEL_CHANGED, PROC_REF(check_security_level))
 	// Adding an extra break for the sake of presentation
 	if(stamina_damage != 0)
 		offensive_notes = "\nVarious interviewed security forces report being able to beat criminals into exhaustion with only [span_warning("[CEILING(100 / stamina_damage, 1)] hit\s!")]"
 
 	register_item_context()
 
+/obj/item/melee/baton/proc/check_security_level(datum/source, new_level)
+	SIGNAL_HANDLER
+	var/mob/living/user = src.loc
+	if(active && istype(user) && new_level < SEC_LEVEL_RED)
+		to_chat(user, span_alert("Your baton turns off as the threat to the station has passed"))
```

Comment:
Does this mean I could toggle to red alert, turn on all my batons, and then flip it back and have the only active batons on the station?
Also why is this off /baton and not /baton/security? do you intend this to apply to telescoping batons?